integration-tests/scaffolding/helpers.ts

```diff
@@ -1,22 +1,166 @@
+            if (result.dispatchError) {
+                let err: EventError;
+
+                switch (result.dispatchError.type) {
+                    case 'Module': {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+
+                    default: {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+                }
+
+                console.log(err.toString());
+                throw err;
+            }
+        }),
+        map((result: ISubmittableResult) => result.events.reduce((acc, { event }) => { acc[eventKey(event)] = event.data; return acc }, {})),
+        // tap((events) => console.log(events)),
+    );
 }
 
-export function eventKey(event:Event): string {
+export function eventKey(event: Event): string {
     return `${event.section}.${event.method}`;
 }
 
 export function signPayloadSr25519(keys: KeyringPair, data: Codec): Sr25519Signature {
-    return {Sr25519: u8aToHex(keys.sign(u8aWrapBytes(data.toU8a())))}
-}
\ No newline at end of file
+    return { Sr25519: u8aToHex(keys.sign(u8aWrapBytes(data.toU8a()))) }
+}
+
+export function createAccount(name: string = 'first pair') {
+    const mnemonic = mnemonicGenerate();
+    // create & add the pair to the keyring with the type and some additional
+    // metadata specified
+    const keyring = new Keyring({ type: 'sr25519' });
+    const keypair = keyring.addFromUri(mnemonic, { name }, 'sr25519');
+
+    return keypair;
+}
+
+export async function createAndFundAccount({ api, amount, source, name, context }: AccountFundingInputs) {
+    const keypair = createAccount(name);
+
+    // Get keypair for pre-funded dev account
+    const devKeys = createKeys(source);
+
+    // Transfer funds from pre-funded dev account to new account
+    await transferFunds(api, devKeys, keypair, amount);
+
+    if (doTxAccounting) {
+        let map = balancesMap.get(context || 'undefined');
+        if (!map) {
+            map = new Map<string, BeginEndBalances>();
+        }
+
+        map.set(keypair.address, { beginBalance: amount });
+        balancesMap.set(context || 'undefined', map);
+    }
+
+    return { newAccount: keypair, devAccount: devKeys };
+}
+
+export async function showTotalCost(api: ApiRx, context?: string) {
+    let cum = 0n;
+    const titleStr = context ? `[${context}]: ` : '';
+    const map = balancesMap.get(context || 'undefined');
+    if (!map) {
+        return;
+    }
+    console.log(`${titleStr}Operation costs were: `);
+    for (const [key, value] of map) {
+        const acct = await getAccountInfo(api, key);
+        value.endBalance = acct.data.free.toBigInt();
+        const cost = value.beginBalance - value.endBalance;
+        cum += cost;
+        console.log(`${titleStr}${key}: ${cost}`);
+    }
+    console.log(`${titleStr}Total cost of all operations was (excluding standard dev accounts): ${cum}`);
```

Comment:
Nicer numbers:

```suggestion
        console.log(`${titleStr}${key}: ${cost.toLocaleString()}`);
    }
    console.log(`${titleStr}Total cost of all operations was (excluding standard dev accounts): ${cum.toLocaleString()}`);
```

> Operation costs were:
5FU2Mxp6AqU5BcCizsD7w4ENqzECfYxd8PVKd1WAHkXEvm1s: 10,913,296
5DkQGNFiU6jzc2MTwsDRWJn7j1QzSvgPtfZXRmcY7p3UPG8t: 1,486,738
5DFieHsHhn5kgXaHfXYCagjDzLVuicCTMPmrSaYScj1Sn8Ay: 9,407,504
5HKRtZEYhrsDVkVKxunRqErjDaFTdfAtxVzSMgfyYGdjXRVY: 7,080,948

Reply:
When formatting, should I scale down 10^-8 to show actual fractionalized token values? Or simply format with commas?